File: app/src/main/java/com/akari/ppx/xp/hook/code/purity/HistoryHook.java
package com.akari.ppx.xp.hook.code.purity;

import com.akari.ppx.xp.hook.code.SuperbHook;

import java.util.List;

import de.robv.android.xposed.XC_MethodReplacement;

import static com.akari.ppx.common.constant.Prefs.DISABLE_HISTORY_ITEMS;

public class HistoryHook extends SuperbHook {
	@Override
	protected void onHook(ClassLoader cl) {
		hookMethod(DISABLE_HISTORY_ITEMS, "com.sup.superb.feedui.repo.a", "a", List.class, XC_MethodReplacement.returnConstant(true));
	}
}

File: app/src/main/java/com/akari/ppx/xp/hook/purity/HistoryHook.kt
@file:Suppress("unused")

package com.akari.ppx.xp.hook.purity

import com.akari.ppx.utils.replaceMethod
import com.akari.ppx.xp.Init.historyPoster
import com.akari.ppx.xp.Init.historyPosterClass
import com.akari.ppx.xp.hook.SwitchHook

class HistoryHook : SwitchHook("disable_history_items") {
    override fun onHook() {
        historyPosterClass!!.replaceMethod(historyPoster(), List::class.java) { true }
    }
}